### PR TITLE
Add a development server w/ watch support, ESBuild and TypeScript

### LINKED ISSUE #4: set up TypeScript and build system

### PR DESCRIPTION
**Pull request by beverloo**
Fixes #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ For example, you could run the following and then access Llaminator at
 `http://localhost:4629`:
 
 ```
-python3 -m http.server 4629
+npm run-script serve
```
Comment (glennhartmann) [inline README.md:26]: we'll also want to add links to how to install node/npm as well as required versions. Do we want something like this [Node CI](https://github.com/GoogleChromeLabs/bubblewrap#bubblewrap) thing? @andreban

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    "module": "esnext",
+    "moduleResolution": "node",
+    "esModuleInterop": false,
+    "noEmitOnError": true,
+    "lib": ["esnext", "dom", "dom.iterable"],
+    "strict": true,
+    "allowSyntheticDefaultImports": true,
+    "baseUrl": "./",
+    "rootDir": "./",
+    "outDir": ".tmp/ts",
```
Comment (glennhartmann) [inline tsconfig.json:13]: we should add this to the gitignore

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "compilerOptions": {
+    "target": "ES2019",
+    "module": "esnext",
+    "moduleResolution": "node",
+    "esModuleInterop": false,
+    "noEmitOnError": true,
+    "lib": ["esnext", "dom", "dom.iterable"],
+    "strict": true,
+    "allowSyntheticDefaultImports": true,
```
Comment (glennhartmann) [inline tsconfig.json:10]: this one seems confusing IMHO. In [this example](https://www.typescriptlang.org/tsconfig#allowSyntheticDefaultImports), I would expect it to import a `React` object from "react", not import the defaults.

File: src/index.ts [hunk 4]
```diff
@@ -53,27 +53,29 @@ window.addEventListener('load', () => {
...
-async function onFileInputChange(e, dbOpenRequest, imgElement, shareBtn) {
-  console.log(e.target.value);
-  if (e.target.files.length === 0) return;
-  const f = e.target.files[0]; // TODO: null-check
+async function onFileInputChange(e: Event, dbOpenRequest: IDBOpenDBRequest, imgElement: HTMLImageElement, shareBtn: HTMLButtonElement) {
+  const inputElement = e.target as HTMLInputElement;
+
+  console.log(inputElement.value);
+  if (!inputElement.files || !inputElement.files.length) return;
+  const f = inputElement.files[0]; // TODO: null-check
```
Comment (glennhartmann) [inline src/index.ts:78]: is it not possible for `f` to be null here? I would've expected typescript to complain

**Review by glennhartmann — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (andreban) [reply README.md:26]: Yes, but we will probably run the tests via Puppeteer. There's a nice article at https://www.freecodecamp.org/news/continuous-integration-with-github-actions-and-puppeteer/.

File: src/index.ts (hunk 4 above)
Comment (andreban) [reply src/index.ts:78]: The TypeScript definition doesn't allow null to be returned when accessing files by index: https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L5224-L5228. Seems to make sense from the spec? https://w3c.github.io/FileAPI/#filelist-section

File: README.md (hunk 1 above)
Comment (beverloo) [reply README.md:37]: Added a bit to the README. We don't yet have dependencies on particular versions of those tools - when we do, that itself can be expressed in package.json.

File: src/index.ts (hunk 4 above)
Comment (beverloo) [reply src/index.ts:78]: I've avoided making changes in this PR unrelated to enabling TypeScript - thus the length check. I'm not seeing where Blink would return a NULL entry in the array.

File: tsconfig.json (hunk 3 above)
Comment (beverloo) [reply tsconfig.json]: No preference, I've removed it for now. Some packages (used to) depend on it.

File: tsconfig.json (hunk 2 above)
Comment (beverloo) [reply tsconfig.json:12]: Done.

**Review by glennhartmann — APPROVED** [review]

**Pull request merged**